### ISSUE TITLE
Non-ASCII characters in drafts are not displayed correctly in thread-view mode and inbox-mode

### ISSUE THREAD

**Issue by rakoo** [sup-heliotrope/sup#343]
From mailing list:

Hello all

I hope this issue hasn't been reported yet. I don't have access to a browser so I couldn't look on github, I went on irc though and no one had seen this issue before. Thanks a lot to sankey who took the time to check github issues for me.

The issue appears in inbox-mode and thread-view-mode for drafts. I haven't seen it for proper mails. Accented characters are displayed as filled in losanges with a question mark inside. There are two losanges per accented character. This is typical of encoding problems. In the editor and in resume-mode, the characters are displayed properly. Japanese and Korean glyphs are also affected.

Steps to reproduce:
Compose a message with accented characters in the subject or at the beginning of mail. Save as draft and go back to inbox mode. Observe weird characters. Go to thread-view-mode. Observe weird characters again.

If there's any questions, I'm also present on the irc channel as guiniol
## 

guiniol

**Comment by danc86** [comment]
Non-ASCII characters in draft message subject lines appear fine now (this may have been fixed by commit c492bac842cd51193ce4e21f042b6998d220e3a3).

But this bug still exists for non-ASCII characters in the draft message body.

**Comment by danc86** [comment]
When the message is still a draft, it won't have any `Content-Type` header by default. From that point of view, Sup's thread view is correct in rendering it as though it must be plain 7-bit ASCII (and thus treats the high bytes as undecodable).

If I edit the draft and manually fill in a `Content-type: text/plain; charset=UTF-8` header, then Sup interprets the draft as UTF-8 and displays it correctly in thread view.

Maybe the bug is really that Sup should be filling in this `Content-Type` header by default in newly created drafts. As of now, the `Content-Type` is only filled in when sending the message, not before.
